utils/authorization_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/mattermost/mattermost-server/model"
+)
+
+type RoleState struct {
+	RoleName   string `json:"roleName"`
+	Permission string `json:"permission"`
+	ShouldHave bool   `json:"shouldHave"`
+}
+
+func mockConfig() *model.Config {
+	config := model.Config{}
+	config.SetDefaults()
+	return &config
+}
+
+func mapping() map[string]map[string][]RoleState {
+
+	policiesRolesMapping := make(map[string]map[string][]RoleState)
+
+	raw, err := ioutil.ReadFile("./policies-roles-mapping.json")
+	if err != nil {
+		panic(err)
+	}
+
+	var f map[string]interface{}
+	err = json.Unmarshal(raw, &f)
+	if err != nil {
+		panic(err)
+	}
+
+	for policyName, value := range f {
+
+		capitalizedName := fmt.Sprintf("%v%v", strings.ToUpper(policyName[:1]), policyName[1:])
+		policiesRolesMapping[capitalizedName] = make(map[string][]RoleState)
+
+		for policyValue, roleStatesMappings := range value.(map[string]interface{}) {
+
+			var roleStates []RoleState
+			for _, roleStateMapping := range roleStatesMappings.([]interface{}) {
+
+				roleStateMappingJSON, _ := json.Marshal(roleStateMapping)
+				var roleState RoleState
+				_ = json.Unmarshal(roleStateMappingJSON, &roleState)
+
+				roleStates = append(roleStates, roleState)
+
+			}
+
+			policiesRolesMapping[capitalizedName][policyValue] = roleStates
+
+		}
+
+	}
+
+	return policiesRolesMapping
+}
+
+func TestSetRolePermissionsFromConfig(t *testing.T) {
+
+	mapping := mapping()
+
+	for policyName, v := range mapping {
+		for policyValue, rolesMappings := range v {
+
+			config := mockConfig()
+			updateConfig(config, policyName, policyValue)
+			roles := model.MakeDefaultRoles()
+			SetRolePermissionsFromConfig(roles, config, true)
+
+			for _, roleMappingItem := range rolesMappings {
+				role := roles[roleMappingItem.RoleName]
+
+				permission := roleMappingItem.Permission
+				hasPermission := roleHasPermission(role, permission)
+
+				if (roleMappingItem.ShouldHave && !hasPermission) || (!roleMappingItem.ShouldHave && hasPermission) {
+					wording := "not to"
+					if roleMappingItem.ShouldHave {
+						wording = "to"
+					}
+					t.Errorf("Expected '%v' %v have '%v' permission when '%v' is set to '%v'.", role.Name, wording, permission, policyName, policyValue)
+				}
+
+			}
+
+		}
+	}
+}
+
+func updateConfig(config *model.Config, key string, value string) {
+	v := reflect.ValueOf(config.ServiceSettings)
+	field := v.FieldByName(key)
+	if !field.IsValid() {
+		v = reflect.ValueOf(config.TeamSettings)
+		field = v.FieldByName(key)
+	}
+	field.Elem().SetString(value)
```

Comment:
@ccbrown Could setting the config value like this cause issues with tests running in parallel?

Reply:
This is only used with config objects created by `mockConfig`, right? Are multiple tests sharing any of those objects?

Reply:
I don't see any issue then.